src/main.rs

```diff
@@ -45,9 +45,13 @@ impl State {
         // into systemd and the session?
         self.ready.call_once(|| {
             // potentially tell systemd we are setup now
-            #[cfg(feature = "systemd")]
             if let state::BackendData::Kms(_) = &self.backend {
+                #[cfg(feature = "systemd")]
                 systemd::ready(&self.common);
```

Comment:
I was thinking about replacing the `systemd::ready` entirely with `dbus::ready`, I think that should work as well. What's good about the dbus solution is not calling external binaries. For now I didn't remove `systemd::ready` in case you consider that too intrusive. I would love to hear your thoughts here.

What I can confirm is that this patch fixes flatpak Steam on my Gentoo+openrc installation.

Reply:
The systemd environment is still something separate from the dbus environment, the dbus-broker just happens to import it on systems with systemd.

Just updating the dbus activation environment would this break e.g. systemd user services. I think this is a sensible solution. If calling external binaries would really be a problem, we could also trigger the systemd interface via dbus, but given this is a one time startup thing, I wouldn't bother.

Reply:
Makes sense, thanks!